packages/backend/src/managers/llama-stack/llamaStackManager.ts

```diff
@@ -0,0 +1,251 @@
+/**********************************************************************
+ * Copyright (C) 2025 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type { TaskRegistry } from '../../registries/TaskRegistry';
+import {
+  type TelemetryLogger,
+  containerEngine,
+  type ContainerProviderConnection,
+  type ContainerCreateOptions,
+} from '@podman-desktop/api';
+import type { PodmanConnection, PodmanConnectionEvent } from '../podmanConnection';
+import llama_stack_images from '../../assets/llama-stack-images.json';
+import { getImageInfo } from '../../utils/inferenceUtils';
+import type { ContainerRegistry, ContainerEvent } from '../../registries/ContainerRegistry';
+import { DISABLE_SELINUX_LABEL_SECURITY_OPTION } from '../../utils/utils';
+import { getRandomName } from '../../utils/randomUtils';
+import type { LlamaStackContainerInfo } from '@shared/models/llama-stack/LlamaStackContainerInfo';
+import { LLAMA_STACK_CONTAINER_TRACKINGID } from '@shared/models/llama-stack/LlamaStackContainerInfo';
+import type { LlamaStackContainerConfiguration } from '@shared/models/llama-stack/LlamaStackContainerConfiguration';
+import path from 'node:path';
+import fs from 'node:fs/promises';
+import type { ConfigurationRegistry } from '../../registries/ConfigurationRegistry';
+import { getFreeRandomPort } from '../../utils/ports';
+
+export const LLAMA_STACK_CONTAINER_LABEL = 'ai-lab-llama-stack-container';
+export const LLAMA_STACK_API_PORT_LABEL = 'ai-lab-llama-stack-api-port';
+
+export class LlamaStackManager {
+  #initialized: boolean;
+  #containerInfo: LlamaStackContainerInfo | undefined;
+
+  constructor(
+    private readonly appUserDirectory: string,
+    private taskRegistry: TaskRegistry,
+    private podmanConnection: PodmanConnection,
+    private containerRegistry: ContainerRegistry,
+    private configurationRegistry: ConfigurationRegistry,
+    private telemetryLogger: TelemetryLogger,
+  ) {
+    this.#initialized = false;
+    this.podmanConnection.onPodmanConnectionEvent(this.watchMachineEvent.bind(this));
+    this.containerRegistry.onStartContainerEvent(this.onStartContainerEvent.bind(this));
+    this.containerRegistry.onStopContainerEvent(this.onStopContainerEvent.bind(this));
```

Comment:
feels like this should be in an init, moreover we should probably have LlamaStackManager implementing Disposable, and dispose those events in it.